src/il_representations/scripts/dqn_train.py

```diff
@@ -0,0 +1,195 @@
+    n_trans = None
+    assert sum([n_config is None for n_config in [n_trajs, n_trans]]) != 0, \
+    'Specify one or none of n_trajs and n_trans, not both.'
+    torch_num_threads = None
+    # the number of 'epochs' is used by the LR scheduler
+    # (we still do `n_batches` total training, the scheduler just gets a chance
+    # to update after every `n_batches / nominal_num_epochs` batches)
+    nominal_num_epochs = 10
+    save_every_n_batches = int(n_batches / nominal_num_epochs)
+    postproc_arch = ()
+    optimizer_class = Adam
+    learning_rate = 1e-3
+    # In SB3's DQNPolicy's implementation, learning rate is not specified as a
+    # optimizer_kwarg, but through lr_schedule
+    optimizer_kwargs = dict()
+
+
+@dqn_train_ex.capture
+def do_training_dqn(venv_chans_first, dict_dataset, out_dir, augs, n_batches,
+                    device_name, final_pol_name, freeze_encoder, postproc_arch,
+                    encoder_path, encoder_kwargs, nominal_num_epochs,
+                    save_every_n_batches, optimizer_class, optimizer_kwargs,
+                    learning_rate, batch_size, n_trans):
+    observation_space = venv_chans_first.observation_space
+    lr_schedule = lambda _: learning_rate
+    device = get_device("auto" if device_name is None else device_name)
+    policy = make_policy(observation_space=observation_space,
+                         action_space=venv_chans_first.action_space,
+                         postproc_arch=postproc_arch,
+                         freeze_pol_encoder=freeze_encoder,
+                         policy_class=CnnPolicy,
+                         encoder_path=encoder_path,
+                         encoder_kwargs=encoder_kwargs,
+                         optimizer_class=optimizer_class,
+                         optimizer_kwargs=optimizer_kwargs,
+                         lr_schedule=lr_schedule).to(device)
+
+    color_space = auto_env.load_color_space()
+    augmenter = augmenter_from_spec(augs, color_space)
+    dataset_length = len(dict_dataset['obs']) if n_trans is None else n_trans
+    progress_df = pd.DataFrame()
+
+    trainer = DQN(
+        policy='CnnPolicy',
+        env=venv_chans_first,
+        device=device_name,
+        buffer_size=dataset_length,
+        batch_size=batch_size,
+        optimize_memory_usage=True
+    )
+    trainer.policy = policy
+
+    # Push data into DQN agent's memory.
+    print('Loading data...')
+    for idx in range(dataset_length):
+        if idx % 1000 == 0:
+            print(f'Loading {idx}/{dataset_length}...')
+        obs, next_obs, action, reward, done = dict_dataset['obs'][idx], \
+                                              dict_dataset['next_obs'][idx], \
+                                              dict_dataset['acts'][idx], \
+                                              dict_dataset['rews'][idx], \
+                                              dict_dataset['dones'][idx]
+        # obs = preprocess_obs(th.tensor(obs),
```

Comment:
Why is/isn't this part necessary?

Reply:
This is to perform image augmentation. I have modified this part and added some comments.